### PR TITLE
feat!: Drop Rails 6.0 EOL

### PR DESCRIPTION
**Pull request by arielvalentin**
6.0 is no longer receiving maintenance, security, or feature updates as of 01 Jun 2023

Users who want to continue instrumentating Rails applications should pin to earlier versions of the instrumentation.

### CODE REVIEW

**Comment by simi** [comment]
:+1:

File: README.md [hunk 1]
```diff
@@ -65,6 +65,19 @@ This repository also contains libraries to aid with interoperablity with vendor
...
 OpenTelemetry Ruby follows the [versioning and stability document][otel-versioning] in the OpenTelemetry specification. Notably, we adhere to the outlined version numbering exception, which states that experimental signals may have a `0.x` version number.
 
+### Library Compatability
+
+This project is managed on a volunteer basis and therefore we have limited capacity to support compatability with unmaintained or EOL libraries.
+
+We will regularly review the instrumentations to drop compatability for any versions of Ruby or gems that reach EOL or no longer receive regular maintenance.
+
+Should you need instrumentation for _older_ versions of a library then you must pin to a specific version of the instrumentation that supports it,
+however you will no longer receive any updates for the instrumentationfrom this repository.
```
Comment (kaylareopelle) [inline README.md:75]:
```suggestion
however, you will no longer receive any updates for the instrumentation from this repository.
```

File: instrumentation/README.md [hunk 2]
```diff
@@ -37,6 +37,15 @@ Instrumentation-specific documentation can be found in each subdirectory's `READ
 
 You also have the option of installing all of the instrumentation libraries by installing `opentelemetry-instrumentation-all`.  See that gem's [README](https://github.com/open-telemetry/opentelemetry-ruby-contrib/tree/main/instrumentation/all) for more.
 
+### Maintenance and Version Compatability
+
+We are a community of volunteers who do our best to provide our users with up to date support for instrumentations,
+however we have limited capacity and are unable to support compatability with EOL or unmaintained libraries.
+
+Should you need to instrument an _older_ version of a library you will have to ensure to pin to a instrumtation version that is compatable with that library.
```
Comment (kaylareopelle) [inline instrumentation/README.md:45]:
```suggestion
Should you need to instrument an _older_ version of a library you will have to ensure to pin to an instrumentation version that is compatible with that library.
```

File: instrumentation/active_record/lib/opentelemetry/instrumentation/active_record/instrumentation.rb [hunk 3]
```diff
@@ -22,11 +21,7 @@ class Instrumentation < OpenTelemetry::Instrumentation::Base
         end
 
         compatible do
-          # We know that releases after MAX_MAJOR_VERSION are unstable so we
-          # check the major version number of the gem installed to make sure we
-          # do not install on a pre-release or full release of the latest
-          # if it exceeds the MAX_MAJOR_VERSION version.
-          gem_version >= MINIMUM_VERSION && gem_version.segments[0] <= MAX_MAJOR_VERSION
```
Comment (kaylareopelle) [inline instrumentation/active_record/lib/opentelemetry/instrumentation/active_record/instrumentation.rb:29]: Why did you decide to remove the MAX_MAJOR_VERSION logic?

Comment (arielvalentin) [reply instrumentation/active_record/lib/opentelemetry/instrumentation/active_record/instrumentation.rb:29]: It's the only place we do it and it felt asymmetric

**Review by kaylareopelle — APPROVED** [review]
This looks good to me! 

I think we should talk more about implementing max major version constraints for Rails-related libraries like was recently mentioned in the #otel-ruby Slack and https://github.com/open-telemetry/opentelemetry-ruby-contrib/issues/373

**Review by simi — APPROVED** [review]

**Review by robertlaurin — APPROVED** [review]

**Pull request merged**